src/ethereum/EthereumTools.ts

```diff
@@ -205,6 +213,29 @@ export class EthereumTools implements EdgeCurrencyTools {
     }
     address = contractAddress
 
+    if (
+      parsedUri.protocol === 'ethereum' &&
+      mustValidateChainId &&
+      !address.includes('@')
+    ) {
+      throw new Error(
+        'chainId must be present when using ethereum: prefix on a non-ethereum network'
+      )
+    } else if (address.includes('@')) {
```

Comment:
This `else` isn't needed, since the previous `if` can only throw. Un-indenting this block would make it more obvious that it runs in every case, assuming we pass the error check.